core/templates/components/forms/schema-based-editors/schema-based-custom-editor.component.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for a schema-based editor for custom values.
+ */
+
+import { Component, EventEmitter, forwardRef, Input, Output } from '@angular/core';
+import { AbstractControl, ControlValueAccessor, NG_VALIDATORS, NG_VALUE_ACCESSOR, ValidationErrors, Validator } from '@angular/forms';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'schema-based-custom-editor',
+  templateUrl: './schema-based-custom-editor.component.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => SchemaBasedCustomEditorComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      multi: true,
+      useExisting: forwardRef(() => SchemaBasedCustomEditorComponent),
+    },
+  ]
+})
+export class SchemaBasedCustomEditorComponent
+implements ControlValueAccessor, Validator {
+  @Input() localValue: string;
+  @Output() localValueChange = new EventEmitter();
+  @Input() schema;
+  @Input() form;
+  onChange: (_: string) => void = () => {};
+  onTouch: () => void;
+  onValidatorChange: () => void = () => {};
+
+  // Implemented as a part of ControlValueAccessor interface.
+  registerOnTouched(fn: () => void): void {
+    this.onTouch = fn;
+  }
+
+  // Implemented as a part of ControlValueAccessor interface.
+  registerOnChange(fn: (_: string) => void): void {
+    this.onChange = fn;
+  }
+
+  // Implemented as a part of ControlValueAccessor interface.
+  writeValue(obj: string): void {
```

Comment:
Is the value written always a string? This seems wrong to me for custom-schema editors.

Reply:
Seems like a mistake, changed to type `unknown` as the type changes a lot.